src/gui/qgsrasterformatsaveoptionswidget.cpp

```diff
@@ -98,10 +100,24 @@ QgsRasterFormatSaveOptionsWidget::QgsRasterFormatSaveOptionsWidget( QWidget *par
   connect( mOptionsHelpButton, &QAbstractButton::clicked, this, &QgsRasterFormatSaveOptionsWidget::helpOptions );
   connect( mOptionsValidateButton, &QAbstractButton::clicked, this, [ = ] { validateOptions(); } );
 
-  // create eventFilter to map right click to swapOptionsUI()
-  // mOptionsLabel->installEventFilter( this );
+  // Install an eventFilter to customize the default QLineEdit contextMenu with an added swapOptionsUI action
   mOptionsLineEdit->installEventFilter( this );
-  mOptionsStackedWidget->installEventFilter( this );
+
+  // Use a Custom Context menu for the widget to swap between modes (table / lineedit)
+  setContextMenuPolicy( Qt::CustomContextMenu );
+  connect( this, &QWidget::customContextMenuRequested, this, [this]( QPoint pos )
+  {
+    QMenu menu( this );
+    QString text;
+    if ( mTableWidget->isVisible() )
+      text = tr( "Use simple interface" );
+    else
+      text = tr( "Use table interface" );
```

Comment:
I realise this isn't introduced here, but let's fix the action capitalization too:

```suggestion
      text = tr( "Use Simple Interface" );
    else
      text = tr( "Use Table Interface" );
```